### ISSUE TITLE
OutOfMemoryError after migrating to version 9.4.19.v20190610 from 9.4.18.v20190429

### ISSUE THREAD

**Issue by paskos** [eclipse/jetty.project#3871]
OS: Amazon Linux AMI 2018.03
Jetty version: 9.4.19.v20190610
Java version :
openjdk version "1.8.0_162"
OpenJDK Runtime Environment (build 1.8.0_162-8u162-b12-1~14.04-b12)
OpenJDK 64-Bit Server VM (build 25.162-b12, mixed mode)

Our application uses embedded jetty for websocket.
We just moved to version 9.4.19.v20190610 from 9.4.18.v20190429 without changes in the server code.

During load testing we experience OutOfMemoryError.
After analyzing the heap dump we see around 3 gigs (out of 4) being consumed by a CopyOnWriteArrayList containing 37556 ContainerLifeCycle$Bean which itself contains a WebSocketSession. 
This problem did not occur with version 9.4.18.v20190429.

I attached some screen shots of Eclipse Memory Analysis Tool.

I wonder if some changes in the latest version requires some changes in our setup to clean up the sessions.

Thanks in advance
<img width="650" alt="accumulation_point" src="https://user-images.githubusercontent.com/2422690/61155358-be012b80-a4be-11e9-9c22-5e1e482d4be0.png">
<img width="795" alt="Content_of_copy_on_write_array_list" src="https://user-images.githubusercontent.com/2422690/61155359-be012b80-a4be-11e9-93ac-01ec25526b51.png">

**Comment by lachlan-roberts** [comment]
Looks like this is a problem with the backport of the WebSocket SessionTracker from jetty-10 #3616.

I made some fixes to this for #3835 which did not make it into the 9.4.19 release. So you could try testing the current 9.4.x code to confirm this fixes your issue, and hopefully we can get a 9.4.20 release out soon with the fix.

**Comment by paskos** [comment]
Some additional information that might help. 
My informations weren't totally accurate.
We did modify some code in our load test websocket client code base following the upgrade.

The client extends `org.eclipse.jetty.websocket.api.WebSocketAdapter` and used to manually call `org.eclipse.jetty.websocket.client.WebSocketClient.onSessionClosed(wsSession);`

Jetty 9.4.19 removed `WebSocketClient.onSessionClosed(wsSession);` method. 
So we removed that call to it but did not replace it with something equivalent.

We wonder if that could be the leading cause of WebsocketSessions accumulation in the server.
